pkg/agent/openflow/packetin.go

```diff
@@ -50,16 +51,37 @@ func (c *client) StartPacketInHandler(stopCh <-chan struct{}) {
 	err := c.SubscribePacketIn(uint8(ofprAction), ch)
 	if err != nil {
 		klog.Errorf("Subscribe PacketIn failed %+v", err)
+		return
 	}
+	packetInQueue := workqueue.NewNamed("packetIn")
+	go c.parsePacketIn(packetInQueue, stopCh)
 
-	wait.PollUntil(time.Second, func() (done bool, err error) {
+	wait.PollUntil(time.Millisecond, func() (done bool, err error) {
 		pktIn := <-ch
+		packetInQueue.Add(pktIn)
+		return false, nil
+	}, stopCh)
+	packetInQueue.ShutDown()
+}
+
+func (c *client) parsePacketIn(packetInQueue workqueue.Interface, stopCh <-chan struct{}) {
+	wait.PollUntil(time.Millisecond, func() (done bool, err error) {
```

Comment:
here again I am a bit confused by the use of `PollUntil`. It seems that this would be more natural and would match our k8s controller implementations:
```
for {
    obj, quit := packetInQueue.Get()
    ...
}
```
When `stopCh` is closed, you call `packetInQueue.ShutDown()` anyway so `Get` will return immediately with `quit` set to true.

Reply:
Fixed